src/main/webapp/script.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Event processing for upload button
+ */
+function fileUploadDisplay() {
+  document.querySelector('.bg-model').style.display = 'flex';
+}
+
+/**
+ * Event processing for close button
+ */
+function closeDiv() {
+  document.querySelector('.bg-model').style.display = 'none';
+}
+
+/**
+ * Makes the form for uploading an image visible and adds the 'action' to it by fetching
+ * the url(that the form needs to post to) from the servlet
+ */
+function fetchBlobstoreUrlAndShowForm() {
+  fetch('/blobstore-upload-url')
+      .then((response) => response.text())
+      .then((imageUploadUrl) => {
+        const uploadForm = document.getElementById('upload-barcode-form');
+        uploadForm.action = imageUploadUrl;
+        uploadForm.classList.remove('hidden');
+      });
```

Comment:
I recently learned that async/await is considered better practice for internal code. Not sure what your style guide says, but using it async functions consistently from the beginning.

Reply:
I tried using async and await like you suggested but the `fetchBlobstoreUrlAndShowForm()` sometimes works and sometimes not. Its not throwing an error or anything.
I am not too sure about how to use async and await but I followed [this example](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await#:~:text=The%20await%20keyword%20causes%20the,%3D%20await%20fetch('coffee.) and tried doing it.
Do you mind checking the changed part of the code for async and await, please?